### PR TITLE
Place a warning about the uses of traits in the documentation

### PR DESCRIPTION
**Pull request by mpdude**
I am not sure if all this is :100: correct, but I'd be happy to learn.

### CODE REVIEW

File: docs/en/reference/limitations-and-known-issues.rst [hunk 1]
```diff
@@ -130,10 +130,49 @@ included in the core of Doctrine ORM. However there are already two
...
 -  `Doctrine2 Hierarchical-Structural Behavior <https://github.com/guilhermeblanco/Doctrine2-Hierarchical-Structural-Behavior>`_
 -  `Doctrine2 NestedSet <https://github.com/blt04/doctrine2-nestedset>`_
 
+Using Traits in Entity Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There is no real "official" statement whether or to which extent traits
+are supported in entity or mapped superclasses.
+
+Traits were added in PHP 5.4 more than 10 years ago, but at the same time
```
Comment (greg0ire) [inline docs/en/reference/limitations-and-known-issues.rst:143]: Time flies!

File: docs/en/reference/limitations-and-known-issues.rst [hunk 2]
```diff
@@ -130,10 +130,49 @@ included in the core of Doctrine ORM. However there are already two
...
+more than two years after the initial Doctrine 2 release and the time where
+core components were designed.
+
+In fact, this documentation mentions traits only in the context of
+:doc:`overriding field association mappings in subclasses <tutorials/override-field-association-mappings-in-subclasses>`. Also, the
+tests in the ORM codebase (as of writing) cover traits only in two (!) places,
+namely the aforementioned override and in an edge-case for the deprecated
+"Entity Generator".
```
Comment (greg0ire) [inline docs/en/reference/limitations-and-known-issues.rst]: This paragraph is going to be hard to maintain: we will have to remember to remove this reference to "Entity Generator" when merging up to 3.0.x, and to change it any time we add any thing about traits in the docs. Maybe you should be more vague here?

**Review by greg0ire — COMMENTED** [review]
Nice job exploring this 👍

File: docs/en/reference/limitations-and-known-issues.rst (hunk 2 above)
Comment (mpdude) [reply docs/en/reference/limitations-and-known-issues.rst]: Changed that.

File: docs/en/reference/limitations-and-known-issues.rst [hunk 3]
```diff
@@ -146,8 +146,8 @@ core components were designed.
 In fact, this documentation mentions traits only in the context of
 :doc:`overriding field association mappings in subclasses <tutorials/override-field-association-mappings-in-subclasses>`. Also, the
 tests in the ORM codebase (as of writing) cover traits only in two (!) places,
-namely the aforementioned override and in an edge-case for the deprecated
-"Entity Generator".
+namely the aforementioned override and in an edge-case for a tool that will be
+removed in Doctrine 3.0.
```
Comment (greg0ire) [inline docs/en/reference/limitations-and-known-issues.rst]: I would replace the whole paragraph with "In fact, traits are rarely mentioned in the docs or in the tests. That way there is no divergence between branches.

File: docs/en/reference/limitations-and-known-issues.rst [hunk 4]
```diff
@@ -130,10 +130,49 @@ included in the core of Doctrine ORM. However there are already two
...
 
-
 -  `Doctrine2 Hierarchical-Structural Behavior <https://github.com/guilhermeblanco/Doctrine2-Hierarchical-Structural-Behavior>`_
 -  `Doctrine2 NestedSet <https://github.com/blt04/doctrine2-nestedset>`_
 
+Using Traits in Entity Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There is no real "official" statement whether or to which extent traits
+are supported in entity or mapped superclasses.
```
Comment (derrabus) [inline docs/en/reference/limitations-and-known-issues.rst]: But since this is the official documentation, we're about to make an official statement, aren't we? I find this paragraph irritating.

Comment (mpdude) [reply docs/en/reference/limitations-and-known-issues.rst]: You’re right.

So, what can we say?

“For the time being, traits are not officially supported.” ?

**Comment by mpdude** [comment]
Updated text suggestions, please 👀

**Review by derrabus — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @mpdude !
